hightime/_timedelta.py

```diff
@@ -0,0 +1,296 @@
+        return self._femtoseconds
+
+    @property
+    def yoctoseconds(self):
+        """yoctoseconds"""
+        return self._yoctoseconds
+
+    # Public methods
+
+    def total_seconds(self):
+        """Total seconds in the duration."""
+        return (
+            (self.days * 86400)
+            + self.seconds
+            + (self.microseconds / 10 ** 6)
+            + (self.femtoseconds / 10 ** 15)
+            + (self.yoctoseconds / 10 ** 24)
+        )
+
+    # String operators
+
+    def __repr__(self):
+        # Follow newer repr: https://github.com/python/cpython/pull/3687
+        r = "{}.{}".format(self.__class__.__module__, self.__class__.__qualname__)
+        r += "("
+        r += ", ".join(
+            "{}={}".format(name, getattr(self, name))
+            for name in _FIELD_NAMES
+            if getattr(self, name) != 0
+        )
+        r += ")"
+        return r
+
+    def __str__(self):
+        s = super().__str__()
+        if self.femtoseconds or self.yoctoseconds:
+            if not self.microseconds:
+                s += "." + "0" * 6
+
+            s += "{:09d}".format(self.femtoseconds)
+
+            if self.yoctoseconds:
+                s += "{:09d}".format(self.yoctoseconds)
+
+        return s
+
+    # Comparison operators
+
+    def __eq__(self, other):
+        if isinstance(other, std_datetime.timedelta):
+            return _cmp(timedelta._as_tuple(self), timedelta._as_tuple(other)) == 0
+        else:
+            return False
+
+    def __ne__(self, other):
+        return not (self == other)
+
+    def __lt__(self, other):
+        return self._cmp(other) < 0
+
+    def __le__(self, other):
+        return self._cmp(other) <= 0
+
+    def __gt__(self, other):
+        return self._cmp(other) > 0
+
+    def __ge__(self, other):
+        return self._cmp(other) >= 0
+
+    def __bool__(self):
+        return any(getattr(self, field) for field in _FIELD_NAMES)
+
+    # Arithmetic operators
+
+    def __pos__(self):
+        return self
+
+    def __abs__(self):
+        return -self if self.days < 0 else self
+
+    def __add__(self, other):
+        if isinstance(other, std_datetime.timedelta):
+            return timedelta(
+                **{
+                    field: getattr(self, field) + getattr(other, field, 0)
+                    for field in _FIELD_NAMES
+                }
+            )
+        return NotImplemented
+
+    __radd__ = __add__
+
+    def __sub__(self, other):
+        if isinstance(other, std_datetime.timedelta):
+            return timedelta(
+                **{
+                    field: getattr(self, field) - getattr(other, field, 0)
+                    for field in _FIELD_NAMES
+                }
+            )
+        return NotImplemented
+
+    def __neg__(self):
+        return timedelta(
+            **{field: -(getattr(self, field, 0)) for field in _FIELD_NAMES}
+        )
+
+    def __mul__(self, other):
+        if isinstance(other, (int, float)):
```

Comment:
Will this exclude "int-like" others?

Reply:
Yes it will, but that's consistent with the standard library implementation.

Reply:
That's interesting. I tested the code with `numpy.int32` in my debugger and found that it _does_ work with int-like types. The method is first called with numpy int, `NotImplemented` is returned, then the method is again called with built-in int!!

Not sure where this magic is baked in, but makes sense for it to be handled so that int-like types don't need to handled explicitly everywhere.